lib/content_publisher/document_collection_group_membership_migrator.rb

```diff
@@ -0,0 +1,28 @@
+module ContentPublisher
+  class DocumentCollectionGroupMembershipMigrator
+    attr_reader :document
+
+    def initialize(document)
+      @document = document
+    end
+
+    def call
+      edition = document.published_edition.present? ? document.published_edition : document.latest_edition
+
+      document_collection_group_memberships = DocumentCollectionGroupMembership.where(document_id: document.id)
+
+      return unless document_collection_group_memberships.any?
+
+      non_whitehall_link = DocumentCollectionNonWhitehallLink.create!(
+        content_id: document.content_id,
+        title: edition.title,
+        base_path: Whitehall.url_maker.public_document_path(edition),
+        publishing_app: "content-publisher",
+      )
+
+      document_collection_group_memberships.each do |membership|
+        membership.update!(document_id: nil, non_whitehall_link_id: non_whitehall_link.id)
+      end
```

Comment:
```suggestion
DocumentCollectionGroupMembership.where(document_id: document.id).update_all(document_id: nil, non_whitehall_link_id: non_whitehall_link.id)
```
We can do these lots of queries in a single query.